### PR TITLE
Dev/nosami/fix 1464804

### PR DESCRIPTION
**Pull request by nosami**
We should get the latest changes into the main branch :)

### CODE REVIEW

File: XtermSharp.Mac/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -4,20 +4,20 @@
...
+//[assembly: AssemblyCopyright ("${AuthorCopyright}")]
+//[assembly: AssemblyTrademark ("")]
+//[assembly: AssemblyCulture ("")]
 
 // The assembly version has the format "{Major}.{Minor}.{Build}.{Revision}".
 // The form "{Major}.{Minor}.*" will automatically update the build and revision,
 // and "{Major}.{Minor}.{Build}.*" will update just the revision.
 
-[assembly: AssemblyVersion ("1.0.*")]
+//[assembly: AssemblyVersion ("1.0.0.0")]
```
Comment (sandyarmstrong) [inline XtermSharp.Mac/Properties/AssemblyInfo.cs:20]: Why not just delete the file?

**Review by sandyarmstrong — APPROVED** [review]

File: XtermSharp/Pty.cs [hunk 2]
```diff
@@ -47,8 +49,71 @@ public static int ForkAndExec (string programName, string [] args, string [] env
...
+				// And then the array itself
+				Marshal.FreeHGlobal ((IntPtr)arr);
+			}
+		}
+
 		[DllImport ("libc", SetLastError = true)]
-		extern static int ioctl (int fd, long cmd, ref UnixWindowSize WinSz);
+		extern static int ioctl (int fd, ulong cmd, ref UnixWindowSize WinSz);
+
+		[DllImport ("libpty.dylib", EntryPoint = "set_window_size")]
```
Comment (Therzok) [inline XtermSharp/Pty.cs:122]:
```suggestion
		[DllImport ("libpty.dylib", EntryPoint = "set_window_size", SetLastError=true)]
```

File: XtermSharp.Mac/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (sgmunn) [reply XtermSharp.Mac/Properties/AssemblyInfo.cs:20]: that was probably me just trying to get things to build from xtermsharp

**Review by sgmunn — APPROVED** [review]

**Review by Therzok — APPROVED** [review]

**Comment by mrward** [comment]
Can we merge this? It looks like vsmac's main branch is not using these fixes but is using the latest commit from the master branch so start without debugger can kill the IDE if the debugged process is stopped.

**Comment by sgmunn** [comment]
> [XtermSharp @ e4a3e8b](https://github.com/xamarin/XtermSharp/tree/e4a3e8b5e35d13098c3763cc860c6aeeb71f6a16)

I think both main and release-17 are pointing to the latest commit here, phew. but yes, this should be merged - unless we want to be able to send these upstream to Miguels original repo and we track our own "main" branch in here?

**Comment by mrward** [comment]
Right now VS Mac from the main branch terminates when you start without the debugger. So I am bit confused about what is fixed where :)

**Comment by nosami** [comment]
Merged!

**Pull request merged**